tests/test_love_producer_base.py

```diff
@@ -0,0 +1,376 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import json
+import asyncio
+import unittest
+import logging
+
+import numpy as np
+
+from astropy.time import Time
+
+from love.producer import LoveProducerBase
+from love.producer.test_utils import cancel_task
+
+
+class TestLoveProducerBase(unittest.IsolatedAsyncioTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.log = logging.getLogger(__name__)
+        cls.standard_timeout = 5
+        cls.time_pooling = 0.1
+
+    async def asyncSetUp(self):
+        self.producer = LoveProducerBase()
+        self.messages_received = []
+        self._run_data_generator_loop = False
+        self.data_generator_task = asyncio.Future()
+        self.data_generator_task.set_result(True)
+        self.generate_async_data_callback = None
+        self.sample_message_data_from_manager = dict(
+            category="initial_state",
+            data=[
+                {"csc": "Test", "salindex": 1, "stream": {"event_name": "summaryState"}}
+            ],
+        )
+        self.sample_summary_state = dict(summaryState=4)
+
+    async def asyncTearDown(self):
+        self.stop_data_generation_loop()
+        try:
+            await asyncio.wait_for(
+                self.data_generator_task, timeout=self.standard_timeout
+            )
+        except asyncio.TimeoutError:
+            self.log.debug(
+                "Timeout waiting for data generator task to complete. Cancelling it."
+            )
+            await cancel_task(self.data_generator_task)
+        except asyncio.CancelledError:
+            pass
+
+    def test_get_initial_state_as_json_no_component_name(self):
+
+        with self.assertRaises(AssertionError):
+            self.producer.get_initial_state_as_json()
+
+    def test_get_initial_state_as_json(self):
+
+        component_name = "Test"
+
+        self.producer.component_name = component_name
+
+        initial_state_subscribe_msg = json.loads(
+            self.producer.get_initial_state_as_json()
+        )
+
+        for key, value in [
+            ("option", "subscribe"),
+            ("category", "initial_state"),
+            ("csc", component_name),
+            ("salindex", "all"),
+            ("stream", "all"),
+        ]:
+            self.assertIn(key, initial_state_subscribe_msg)
+            self.assertEqual(initial_state_subscribe_msg[key], value)
+
+    def test_add_metadata(self):
+
+        self.setup_for_data_handling_test()
+
+        self.producer.add_metadata(component_index=1)
```

Comment:
As `self.setup_for_data_handling_test` has `self.producer.add_metadata` I should include the metadata inside the first function.